### ISSUE TITLE
Layer panel - allow users to select/search from a list of unique values for a specified feature service field

### ISSUE THREAD

**Issue by csphang** [wri/gfw-mapbuilder#317]
Support feature layers and dynamic layers.

User specifies the service URL and the field that should be queried in the parameter. Labels use the field alias for display.

### Example
Endpoint URL: https://gis-gfw.wri.org/arcgis/rest/services/partner_requests/greenpeace_en/MapServer/0
Field: po_grup
Field alias: Group

See example from ZSL SPOTT:
![2019-01-22_16-24-35](https://user-images.githubusercontent.com/13664037/51566417-7a5c2c00-1e62-11e9-8aa4-14c6e324f7b4.jpg)

**Comment by jenboyd** [comment]
@csphang link to test: http://alpha.blueraster.io/gfw-mapbuilder/317-layer-filter-fields/

Layer names are `Palm Oil Concessions - feature layer` and `Palm Oil Concessions - dynamic layer`

The dynamic layer can accept a multiple sublayers, but all sublayers would need to have the same filter field.. is this what you are thinking?

**Comment by csphang** [comment]
Continue to support dynamic layers, but will only filter attributes based on one field.

**Comment by richardbarad** [comment]
Hi @csphang, @SwampGuzzler , and @jenboyd.

This looks great! I just shared it with our collague who manages the RSPO mapbuilder app and he is very excited about being able to filter his palm oil concessions by company? https://www.rspo.org/geo-rspo

I took a look at the configuration and had a few questions:

1) Could we make the "filter by group" text adjustable in the configuration? I could see instances where someone might want to "filter by type" or "filter by status" depending on what layer they are filtering.

2) Also, how hard would it be to support multiple languages here? Both, for the text label and for the field. One of the teams which is interested in this functionality is Georgia and thier application is available in multiple languages. 

Richard

**Comment by jenboyd** [comment]
Link updated with comments above: http://alpha.blueraster.io/gfw-mapbuilder/317-layer-filter-fields/

**Comment by jenboyd** [comment]
Example of config for both feature and dynamic layer types
```
 {
        order: 9,
        id: 'PALM_OIL',
        type: 'feature',
        url: 'https://gis-gfw.wri.org/arcgis/rest/services/partner_requests/greenpeace_en/MapServer/0',
        visible: false,
        label: {
          en: 'Palm Oil Concessions - feature layer',
          fr: 'Palm Oil Concessions - feature layer',
          es: 'Palm Oil Concessions - feature layer',
          pt: 'Palm Oil Concessions - feature layer',
          id: 'Palm Oil Concessions - feature layer',
          zh: 'Palm Oil Concessions - feature layer',
          ka: 'Palm Oil Concessions - feature layer'
        },
        sublabel: {
          en: '(incl. rubber, coconut)',
          fr: '(incl. rubber, coconut)',
          es: '(incl. rubber, coconut)',
          pt: '(incl. rubber, coconut)',
          id: '(incl. rubber, coconut)',
          zh: '(incl. rubber, coconut)',
          ka: '(incl. rubber, coconut)'
        },
        filterField: {
          en: 'po_grup',
          fr: 'po_grup',
          es: 'po_grup',
          pt: 'po_grup',
          id: 'po_grup',
          zh: 'po_grup',
          ka: 'po_grup'
        },
        filterLabel: {
          en: 'Filter by group',
          fr: 'Filter by group',
          es: 'Filter by group',
          pt: 'Filter by group',
          id: 'Filter by group',
          zh: 'Filter by group',
          ka: 'Filter by group'
        },
  },  {
        order: 10,
        id: 'PALM_OIL_DYNAMIC',
        type: 'dynamic',
        url: 'https://gis-gfw.wri.org/arcgis/rest/services/partner_requests/greenpeace_en/MapServer',
        layerIds: [0],
        visible: false,
        label: {
          en: 'Palm Oil Concessions - dynamic layer',
          fr: 'Palm Oil Concessions - dynamic layer',
          es: 'Palm Oil Concessions - dynamic layer',
          pt: 'Palm Oil Concessions - dynamic layer',
          id: 'Palm Oil Concessions - dynamic layer',
          zh: 'Palm Oil Concessions - dynamic layer',
          ka: 'Palm Oil Concessions - dynamic layer'
        },
        sublabel: {
          en: '(incl. rubber, coconut)',
          fr: '(incl. rubber, coconut)',
          es: '(incl. rubber, coconut)',
          pt: '(incl. rubber, coconut)',
          id: '(incl. rubber, coconut)',
          zh: '(incl. rubber, coconut)',
          ka: '(incl. rubber, coconut)'
        },
        filterField: {
          en: 'po_grup',
          fr: 'po_grup',
          es: 'po_grup',
          pt: 'po_grup',
          id: 'po_grup',
          zh: 'po_grup',
          ka: 'po_grup'
        },
        filterLabel: {
          en: 'Filter by group',
          fr: 'Filter by group',
          es: 'Filter by group',
          pt: 'Filter by group',
          id: 'Filter by group',
          zh: 'Filter by group',
          ka: 'Filter by group'
        }
}
```

**Comment by richardbarad** [comment]
@jenboyd! Look good!

Just to confirm, are pop-up configurable on these layers and would you configure them the same way as a regular dynamic/feature layer?

**Comment by jenboyd** [comment]
@richardbarad Yep, I did a quick test. It will work the same as it did previously.

**Comment by richardbarad** [comment]
Hi @SwampGuzzler and @csphang,

Just tried testing this in the CMS on our Georgia atlas here:  http://geo2.doesntexist.org/map?l=en

Here is the layer configuration I am using with the filter. Is something incorrect in my configuration?

```
{
        "order": 3,
        "id": "EmeraldNetwork",
        "type": "dynamic",
        "url": "https://gis.mepa.gov.ge/server/rest/services/atlas/Biodiversity/MapServer",
        "layerIds": [
          2
        ],
        "label": {
          "en": "Emerald Network",
          "ka": "ზურმუხტის ქსელი"
        },
        "filterField": {
          "en": "Status_EN",
          "ka": "Status_KA"
        },
        "filterLabel": {
          "en": "Filter by Status",
          "ka": "Filter by Status"
        },
        "popup": {
          "title": {
            "en": "Emerald Network",
            "ka": "ზურმუხტის ქსელი"
          },
          "content": {
            "ka": [
              {
                "label": "სტატუსი",
                "fieldExpression": "Status_KA"
              },
              {
                "label": "საიტის კოდი",
                "fieldExpression": "Sitecode_KA"
              },
              {
                "label": "ფართობი ჰა",
                "fieldExpression": "Area_ha_KA"
              }
            ],
            "en": [
              {
                "label": "Status",
                "fieldExpression": "Status_EN"
              },
              {
                "label": "Sitecode",
                "fieldExpression": "Sitecode_KA"
              },
              {
                "label": "Area ha",
                "fieldExpression": "Area_ha_KA"
              }
            ]
          }
        }
      }
```

**Comment by richardbarad** [comment]
I tested this in two different builds using `https://alpha.blueraster.io.s3.amazonaws.com/gfw-mapbuilder/merge-of-primary-forest-and-2018/1.3.4.js`

I was able to get it to work in the RSPO build (attached).

[RSPO.zip](https://github.com/wri/gfw-mapbuilder/files/3113035/RSPO.zip)

However, I could not get it to work in the Georgia build. Also attached. I think my configuration is correct but let me know if anything looks incorrect. The Georgia build includes multiple languages and special characters so maybe this is the issue?

[GEO_new.zip](https://github.com/wri/gfw-mapbuilder/files/3113042/GEO_new.zip)

**Comment by SwampGuzzler** [comment]
Approved by @richardbarad after testing the latest fix from #370 on this [build](https://alpha.blueraster.io.s3.amazonaws.com/gfw-mapbuilder/317-georgia-filters/external.html)
